optaplanner-core/src/test/java/org/optaplanner/core/api/score/stream/IntervalJoinersTest.java

```diff
@@ -0,0 +1,951 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.optaplanner.core.api.score.stream;
+
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import org.junit.jupiter.api.Test;
+import org.optaplanner.core.api.function.QuadFunction;
+import org.optaplanner.core.api.function.TriFunction;
+import org.optaplanner.core.impl.score.stream.bi.AbstractBiJoiner;
+import org.optaplanner.core.impl.score.stream.penta.AbstractPentaJoiner;
+import org.optaplanner.core.impl.score.stream.quad.AbstractQuadJoiner;
+import org.optaplanner.core.impl.score.stream.tri.AbstractTriJoiner;
+
+public class IntervalJoinersTest {
```

Comment:
All unit tests for the class Joiners should be in JoinersTests normally, so it's easier to find them. It's harder to know that this class exists, because it's not classname +"Test"

Reply:
Issue is the Interval Joiner tests are a lot more complicated than the normal joiner tests as there are 13 test cases we need to go through instead of the typical 2. So adding it to JoinersTests will make it massive: ~800 lines. As we add more and more Joiners, eventually JoinersTests will be a couple thousand lines and get harder and harder to maintain. Since 800 isn't too bad, I'll merge it, but we should separate the tests if we add more joiners (in particular, those with non-trivial testing)